### PR TITLE
Jetpack Manage: Update Boost expandable block to support 'Configure Boost' flow.

### PR DESCRIPTION
**Pull request by jkguidaven**
This PR updates the expandable block to support a 1-time score and render the correct CTA button.

Screen Shot 2023-09-20 at 3 00 27 PM
Screen Shot 2023-09-20 at 2 59 02 PM

Closes https://github.com/Automattic/jetpack-genesis/issues/27

## Proposed Changes

* Update the Jetpack Boost expandable block to render the score regardless of whether the site has an active Jetpack Boost plugin.
* Update the block to display the **'Configure Boost'** button if the site has no active Jetpack boost plugin. The CTA should show the **'Optimize CSS'** button if there is no active plugin.

## Testing Instructions
Since these changes are made specifically for agencies, you must set yourself(partner) as an agency - 2c49b-pb. Make sure to switch it back to the previous type.

### Optimize CSS button
* Spin up a JN site and connect it to the Jetpack cloud. Make sure you install **Jetpack Boost** plugin and generate a score with it.
* Confirm that the Jetpack Boost expandable block displays the score with the **'Optimize CSS'** button.

Screen Shot 2023-09-20 at 2 59 02 PM

### Configure Boost
* Once you have done the above test, uninstall the Jetpack Boost plugin on the JN Site.
* Confirm the Jetpack Boost expandable block will display the **'Configure Boost'** button.
Screen Shot 2023-09-20 at 3 00 27 PM

## Pre-merge Checklist

- [ ] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [ ] [Have you written new tests](https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md) for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [ ] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
- [ ] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-ajp-p2)?

### CODE REVIEW

File: client/jetpack-cloud/sections/agency-dashboard/sites-overview/site-expanded-content/boost-site-performance.tsx
Comment (mattgawarecki) [inline client/jetpack-cloud/sections/agency-dashboard/sites-overview/site-expanded-content/boost-site-performance.tsx]: Non-blocking:

Generally aside from static constants (which are `ALL_CAPS`), components (which are `PascalCase`), and special instances, variables in Calypso conventionally start with a lower-case letter and are `camelCase` (e.g, `optimizeCssUrl`, `configureBoostUrl`).

**Review by mattgawarecki — COMMENTED** [review]
Reading through the code, this all seems to make sense. I left one non-blocking comment.

Functional testing didn't go quite as planned, however: when I make changes to my testing site -- for example, getting a Boost score or uninstalling the plugin -- the dashboard doesn't update to reflect the changes. I suspect this is because we're sourcing data from Site Profiles, which only updates periodically? I'll check back again over the course of today and tomorrow, and if all goes well, I'll approve.

File: client/jetpack-cloud/sections/agency-dashboard/sites-overview/site-expanded-content/test/boost-site-performance.test.tsx
Comment (yashwin) [inline client/jetpack-cloud/sections/agency-dashboard/sites-overview/site-expanded-content/test/boost-site-performance.test.tsx]: I think we could test this further:

- Check if the button has the correct URL.
- Simulate a click and see if `trackEventMock` is called with the correct event name.

File: client/jetpack-cloud/sections/agency-dashboard/sites-overview/site-expanded-content/test/boost-site-performance.test.tsx
Comment (yashwin) [inline client/jetpack-cloud/sections/agency-dashboard/sites-overview/site-expanded-content/test/boost-site-performance.test.tsx:100]: I think we could test this further:

- Check if the button has the correct URL.
- Simulate a click and see if `trackEventMock` is called with the correct event name.

**Review by yashwin — CHANGES_REQUESTED** [review]
I found an issue while testing this.

It shows "Add Boost" even though I have added Boost to the site. 

Screenshot 2023-09-21 at 8 31 33 AM

I think this is due to the overall score being null initially.

Screenshot 2023-09-21 at 8 34 11 AM

Steps to reproduce:

Create a new JN site > Assing Boost license > Expand the site content

File: client/jetpack-cloud/sections/agency-dashboard/sites-overview/site-expanded-content/boost-site-performance.tsx
Comment (yashwin) [inline client/jetpack-cloud/sections/agency-dashboard/sites-overview/site-expanded-content/boost-site-performance.tsx]: I can we can write this as below. WDYT?

```
const buttonProps = hasBoost
  ? {
      label: 'Optimize CSS',
      href: OptimizeCSSUrl,
      onClick: () => trackEvent('expandable_block_optimize_css_click'),
    }
  : {
      label: 'Configure Boost',
      href: ConfigureBoostUrl,
      onClick: () => trackEvent('expandable_block_configure_boost_click'),
    };

return (
  <Button
    href={buttonProps.href}
    target="_blank"
    onClick={buttonProps.onClick}
    className="site-expanded-content__card-button"
    compact
  >
    {translate(buttonProps.label)}
  </Button>
);
```

File: client/jetpack-cloud/sections/agency-dashboard/sites-overview/site-expanded-content/boost-site-performance.tsx
Comment (jkguidaven) [reply client/jetpack-cloud/sections/agency-dashboard/sites-overview/site-expanded-content/boost-site-performance.tsx]: Good catch 👍. I keep missing the camel case for some reason, but I'll keep this in mind. 

I have some changes suggested by Yashwin, which I ended up removing the variables.

File: client/jetpack-cloud/sections/agency-dashboard/sites-overview/site-expanded-content/test/boost-site-performance.test.tsx
Comment (jkguidaven) [reply client/jetpack-cloud/sections/agency-dashboard/sites-overview/site-expanded-content/test/boost-site-performance.test.tsx]: This has been updated 👍

File: client/jetpack-cloud/sections/agency-dashboard/sites-overview/site-expanded-content/test/boost-site-performance.test.tsx
Comment (jkguidaven) [reply client/jetpack-cloud/sections/agency-dashboard/sites-overview/site-expanded-content/test/boost-site-performance.test.tsx:100]: This has been updated 👍

File: client/jetpack-cloud/sections/agency-dashboard/sites-overview/site-expanded-content/boost-site-performance.tsx
Comment (jkguidaven) [reply client/jetpack-cloud/sections/agency-dashboard/sites-overview/site-expanded-content/boost-site-performance.tsx]: Good suggestion. I have applied this to my changes. thanks 👍

**Comment by jkguidaven** [comment]
> I found an issue while testing this.
> 
> It shows "Add Boost" even though I have added Boost to the site.
> 
> Screenshot 2023-09-21 at 8 31 33 AM
> I think this is due to the overall score being null initially.
> 
> Screenshot 2023-09-21 at 8 34 11 AM
> Steps to reproduce:
> 
> Create a new JN site > Assing Boost license > Expand the site content

I initially intended to show the empty content regardless of whether Boost existed. My thought was that we would eventually change the text to mention **'Get Score'** (which we did on a separate PR). 

But after rebasing this PR and seeing the changes come together, I realized it makes more sense to show the scorecard when there is a Boost in the site, even though we don't have the scores yet. This also matches the behavior on the 'Boost' column, which makes the overall behavior coherent.

**Comment by jkguidaven** [comment]
> Functional testing didn't go quite as planned, however: when I make changes to my testing site -- for example, getting a Boost score or uninstalling the plugin -- the dashboard doesn't update to reflect the changes. I suspect this is because we're sourcing data from Site Profiles, which only updates periodically? I'll check back again over the course of today and tomorrow, and if all goes well, I'll approve.

This is also what I noticed. I'm not 100% sure when Site Profiles are reindexed, but there are a couple of instances where we did cache some of the data (e.g., threats). I won't be surprised if we also cache the Site profiles somewhere. I usually wait a few minutes and refresh to get the updated information.

**Comment by jkguidaven** [comment]
@mattgawarecki and @yashwin, thanks for the review. I have addressed the PR comments. Can you have another look if you get a chance?

**Review by yashwin — APPROVED** [review]
Code changes LGTM 👍 Thanks for addressing my comments.

However, I faced a similar issue that @mattgawarecki reported. While that is not related to the changes in the PR, I'm approving it. But, let's wait for his approval too 🙂

**Comment by mattgawarecki** [comment]
Checking back in here to say that I haven't seen any change in the results the UI is showing me; but the "good" news is, we're reflecting exactly the data the API is returning. I'll check again throughout the day to watch for any update in (what I assume to be) Site Profiles data.

**Comment by jkguidaven** [comment]
> Checking back in here to say that I haven't seen any change in the results the UI is showing me; but the "good" news is, we're reflecting exactly the data the API is returning. I'll check again throughout the day to watch for any update in (what I assume to be) Site Profile data.

@mattgawarecki, maybe you can use the Site profile MC tool or install a plugin to force reindexing when changes don't reflect. I'm not sure why your site profile is not updating.

**Review by mattgawarecki — APPROVED** [review]
I'm approving, everything seems to work the way I'd expect. 👍

**Pull request merged**
